foreman.fc

```diff
@@ -34,14 +34,19 @@
 # Passenger non-SCL file contexts
 
 /usr/lib/ruby/gems/1.8/gems/passenger-.*/agents/.* -- gen_context(system_u:object_r:passenger_exec_t,s0)
-/usr/lib64/ruby/site_ruby/1.8/x86_64-linux/agents/.* -- gen_context(system_u:object_r:passenger_exec_t,s0)
+/usr/lib/ruby/site_ruby/1.8/x86_64-linux/agents/.* -- gen_context(system_u:object_r:passenger_exec_t,s0)
+/usr/libexec/passenger/.* -- gen_context(system_u:object_r:passenger_exec_t,s0)
+
 /usr/lib/ruby/gems/1.8/gems/passenger-.*/helper-scripts/prespawn -- gen_context(system_u:object_r:passenger_exec_t,s0)
+/usr/share/passenger/helper-scripts/prespawn -- gen_context(system_u:object_r:passenger_exec_t,s0)
+
 /var/run/rubygem-passenger(.*)?         gen_context(system_u:object_r:passenger_var_run_t,s0)
 
-# Passenger SCL file contexts
+# Passenger SCL file contexts (/opt/*/root prefix is distribution equivalence)
 
-/opt/theforeman/tfm/root/usr/(share/gems/gems|lib64/gems/exts)/passenger-.*/agents/.* -- gen_context(system_u:object_r:passenger_exec_t,s0)
-/opt/theforeman/tfm/root/usr/(share/gems/gems|lib64/gems/exts)/passenger-.*/helper-scripts/prespawn -- gen_context(system_u:object_r:passenger_exec_t,s0)
+/usr/share/gems/gems/passenger-.*/agents/.* -- gen_context(system_u:object_r:passenger_exec_t,s0)
+/usr/share/gems/gems/passenger-.*/helper-scripts/prespawn -- gen_context(system_u:object_r:passenger_exec_t,s0)
+/usr/lib/gems/(exts|ruby)/passenger-.*/agents/.* -- gen_context(system_u:object_r:passenger_exec_t,s0)
```

Comment:
This will need to be duplicated for EL6 and changed to lib64, only EL7 seems to understand (and require) the lib/lib64 equivalence.

<pre>
-rwxr-xr-x. root root system_u:object_r:lib_t:s0       /opt/theforeman/tfm/root/usr/lib64/gems/ruby/passenger-4.0.18/agents/SpawnPreparer
</pre>